lib/src/services/coin_service/app_coin_service.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import '/../src/config/app_config.dart';
+import '/../src/config/constants.dart';
+
+import '/../src/models/coins.dart';
+import '/../src/services/coin_service/coin_service.dart';
+
+class AppAlbumService extends CoinService {
+  AppAlbumService(http.Client client) : super(client);
+
+  @override
+  Future<List<Coins>> fecthCoins() async {
+    Map<String, String> qParams = {
+      AppConfig.instance.getValue(AppConstants.currency):
+          AppConstants.currencyOfMarket,
+      AppConfig.instance.getValue(AppConstants.order): AppConstants.orderBy,
+      AppConfig.instance.getValue(AppConstants.perPage):
+          AppConstants.amountPerPage,
+      AppConfig.instance.getValue(AppConstants.page): AppConstants.numberPage,
+      AppConfig.instance.getValue(AppConstants.sparkline):
+          AppConstants.valueSparkline,
+    };
+
+    final uri = Uri(
+        scheme: 'https',
+        host: AppConfig.instance.getValue(AppConstants.hostName),
+        path: AppConfig.instance.getValue(AppConstants.coinPath) +
+            AppConfig.instance.getValue(AppConstants.marketPath),
+        queryParameters: qParams);
+
+    var response = await client.get(uri);
+
+    if (response.statusCode == 200) {
+      Iterable responseList = json.decode(response.body);
+
+      var coins =
+          List<Coins>.from(responseList.map((model) => Coins.fromJson(model)));
```

Comment:
what happens if fromJson return exception

Reply:
Can I use try catch to it ?